app/modules/site_settings/models.py

```diff
@@ -218,10 +227,13 @@ def set(
         boolean=None,
         override_readonly=False,
     ):
-        if is_extension_enabled('edm') and cls.is_edm_key(key):
-            raise ValueError(
-                f'forbidden to directly set key with prefix "{cls.EDM_PREFIX}" via (key={key})'
-            )
+        if (
```

Comment:
This is certainly possible but I cut off at a point where it was a lot clearer than it was.

Reply:
We can certainly do this in a later PR.  There's validation in `set_key_value` but not in `set`.  Just looking at the method names, it's not really easy to know which one to use

Reply:
I definitely see what you mean, you have a valid point. I hadn't spotted that. 
This change was to tidy up the rats nest of REST related APIs and for the most part I'd left the 'internal houston to site settings' apis alone. 
I'm keen to not add too much more to this PR, it's virtually a complete rewrite of the module as it is but now you've made me aware of this, it may get fixed 'quite soon'.
Also getting this PR in will unblock both EDM retirement and autogenerated names so I'd rather not polish this any more right now.